mlflow/models/utils.py

```diff
@@ -487,24 +532,44 @@ def _is_sparse_matrix(x):
         elif isinstance(pf_input, pd.DataFrame):
             new_pf_input = {}
             for col_name, tensor_spec in zip(input_schema.input_names(), input_schema.inputs):
-                new_pf_input[col_name] = _enforce_tensor_spec(
-                    np.array(pf_input[col_name], dtype=tensor_spec.type), tensor_spec
+                pd_series = pf_input[col_name]
+                new_pf_input[col_name] = _reshape_and_cast_pandas_column_values(
+                    col_name, pd_series, tensor_spec
                 )
         else:
             raise MlflowException(
                 "This model contains a tensor-based model signature with input names, which"
-                " suggests a dictionary input mapping input name to tensor, but an input of"
-                " type {0} was found.".format(type(pf_input))
+                " suggests a dictionary input mapping input name to tensor, or a pandas"
+                " dataframe input containing columns mapping input name to list value flattened"
+                " from tensor, but an input of type {0} was found.".format(type(pf_input))
             )
     else:
+        tensor_spec = input_schema.inputs[0]
         if isinstance(pf_input, pd.DataFrame):
-            new_pf_input = _enforce_tensor_spec(pf_input.to_numpy(), input_schema.inputs[0])
+            num_input_columns = len(pf_input.columns)
+            if num_input_columns == 0:
+                raise MlflowException("Input dataframe contains zero column.")
```

Comment:
```suggestion
                raise MlflowException("Input DataFrame is empty")
```

Reply:
If dataframe is empty, MLFlow@master code will generate `array([], shape=(0, 0), dtype=float64)` instance and pass it to model predict function, this is meaningless and will raise error in any case.